modules/apps/commerce/commerce-pricing-web/src/main/resources/META-INF/resources/commerce_price_lists/cp_instance/js/AddPriceEntryForm.js

```diff
@@ -0,0 +1,347 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import ClayForm, {ClayInput, ClaySelectWithOption} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import ServiceProvider from 'commerce-frontend-js/ServiceProvider/index';
+import {FDS_UPDATE_DISPLAY} from 'commerce-frontend-js/utilities/eventsDefinitions';
+import PropTypes from 'prop-types';
+import React, {useMemo, useRef, useState} from 'react';
+
+import './../../../css/main.scss';
+
+function AddPriceEntryForm({
+	basePrice,
+	cpInstanceId,
+	currency,
+	dataSetId,
+	namespace,
+	precision,
+	priceLists,
+	unitOfMeasures,
+}) {
+	const [inputGroups, setInputGroups] = useState([
+		{
+			id: 'inputGroup-0',
+			price: basePrice,
+			priceListId: (priceLists || [])[0]?.value || 0,
+			unitOfMeasureKey: (unitOfMeasures || [])[0]?.value || null,
+		},
+	]);
+	const inputGroupCounterRef = useRef(0);
+	const hasUnitOfMeasuresRef = useRef(
+		Liferay.FeatureFlags['COMMERCE-11287'] &&
+			!!(unitOfMeasures || []).length
+	);
+
+	const AdminPricingResource = useMemo(
+		() => ServiceProvider.AdminPricingAPI('v2'),
+		[]
+	);
+
+	const closeModal = (modalConfig = {}) => {
+		let openerWindow = Liferay.Util.getOpener();
+
+		openerWindow.Liferay.fire('closeModal', modalConfig);
+
+		if (dataSetId) {
+			do {
```

Comment:
That's because the "opener" of the modal is the primary page and it's different from where is the "dataSet" (it's inside a panel (so in an iframe)). From the _opener_ (the only thing I can use here) I cannot find another way to reach the _panel of the dataSet_ to fire the event